### PR TITLE
Fix authorizer creation by storing on Apigateway store

### PR DESCRIPTION
**Pull request by calvernaz**
Pulumi uses the import API from OpenAPI spec to create the resources. The authoriser was created using the moto "Authorizer" model object and stored on the service backend (moto) instead of the Apigateway store. This discrepancy is related to a recent change to enhance parity that makes "Authorizers" created using the API stored on the Apigateway store.

This PR handles the issue by creating the authorizer through the import api to be stored on the Apigateway store.

Issue https://github.com/localstack/localstack/issues/7391

This PR:
- Moves previous patch method "put_rest_api" into the ApiGatewayProvider handler
- Stores the Authorizer on the Apigateway store `ApiGatewayStore`
- Creates a new module and small snapshot test for apigateway

### CODE REVIEW

File: tests/integration/apigateway/test_apigateway_api.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from localstack.utils.files import load_file
+from tests.integration.apigateway import OPENAPI_SPEC_PULUMI_JSON
+
+
+def test_import_rest_api(import_apigw, snapshot):
```
Comment (calvernaz) [inline tests/integration/apigateway/test_apigateway_api.py:5]: Not too exhaustive for now because we will be moving "import api" tests from existing modules.

File: localstack/services/apigateway/helpers.py [hunk 2]
```diff
@@ -584,8 +586,9 @@ def create_authorizer(path_payload: dict) -> Authorizer:
 
                     if authorizers.get(security_scheme_name):
                         return authorizers.get(security_scheme_name)
-                    authorizer = rest_api.create_authorizer(
-                        create_resource_id(),
+
+                    authorizer = Authorizer(
```
Comment (whummer) [inline localstack/services/apigateway/helpers.py:590]: This is really a bit of an unfortunate "split-brain" situation here (as discussed). We could also avoid using the `moto` models altogether here, and use the ASF generated `Authorizer` typed dict instead. (to avoid a situation where moto's `Authorizer.__init__` may have side effects at some point in the future) 

In any case, over time we need to find a good way to handle this setup (potentially rewriting part of the core, to decouple ourselves from moto..) 👍

File: localstack/services/apigateway/provider.py [hunk 3]
```diff
@@ -644,14 +657,13 @@ def import_rest_api(
             CreateRestApiRequest(name=openapi_spec.get("info").get("title")),
         )
 
-        return _call_moto(
+        return self.put_rest_api(
```
Comment (whummer) [inline localstack/services/apigateway/provider.py:660]: Out of curiosity - was this change required? The `_call_moto(..)` util function is performing a few internal remappings, creating a new local `context`, etc. We would lose that context remapping/rectification if we call the method directly.

File: tests/integration/test_apigateway.py [hunk 4]
```diff
@@ -219,6 +219,7 @@ def test_create_rest_api_with_custom_id(
         assert response.ok
         assert response._content == b'{"echo": "foobar", "response": "mocked"}'
 
+    @pytest.mark.skip
```
Comment (whummer) [inline tests/integration/test_apigateway.py:222]: Is this test currently flaking? If so, can we add a TODO comment and keep a note to fix it in one of the upcoming iterations 👍

**Review by whummer — COMMENTED** [review]
Nice catch, glad we were able to find the issue with authorizers being tracked incorrectly as moto models (over time we'll need to find a good strategy to deal with the split brain situation). 

Changes LGTM in general, just added a few questions / suggestions. Would be good to keep @bentsku in the loop as well 👍

File: tests/integration/test_apigateway_integrations.snapshot.json [hunk 5]
```diff
@@ -1,13 +1,12 @@
 {
   "tests/integration/test_apigateway_integrations.py::test_lambda_proxy_integration": {
```
[the remark this answers is not in the document]
Comment (calvernaz) [reply tests/integration/test_apigateway_integrations.snapshot.json:2]: Yeah Daniel introduce them, but I have updated - tags, version and binary modes were failing so I wonder if that was the reason.

File: localstack/services/apigateway/helpers.py (hunk 2 above)
Comment (calvernaz) [reply localstack/services/apigateway/helpers.py:590]: Next iteration on them will do it.

File: tests/integration/test_apigateway.py (hunk 4 above)
Comment (calvernaz) [reply tests/integration/test_apigateway.py:222]: Just locally doesn't work it all.

File: localstack/services/apigateway/provider.py (hunk 3 above)
Comment (calvernaz) [reply localstack/services/apigateway/provider.py:660]: Good point, I think we can go directly to moto.

File: tests/integration/test_apigateway_integrations.snapshot.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (bentsku) [reply tests/integration/test_apigateway_integrations.snapshot.json:2]: This snapshot looks like it has been updated against LocalStack and not AWS I think?

[the remark this answers is not in the document]
Comment (calvernaz) [reply tests/integration/test_apigateway_integrations.snapshot.json:2]: I can double check, that wasn't the intent tho

[the remark this answers is not in the document]
Comment (bentsku) [reply tests/integration/test_apigateway_integrations.snapshot.json:2]: Of course! There is no provided flow for testing/regenerating snapshot yet, and there's not a defined way. 
Personally I have 2 saved test configurations (`verify`/`update`) in PyCharm and I need to copy/paste the test reference in it to run it, but it saves some headaches. I know @steffyP has a different flow too. We should document some ways to make it easier.

[the remark this answers is not in the document]
Comment (calvernaz) [reply tests/integration/test_apigateway_integrations.snapshot.json:2]: I think because was the first time I just deleted. But at the time it was only AWS, so I will confirm

File: localstack/services/apigateway/provider.py (hunk 3 above)
Comment (whummer) [reply localstack/services/apigateway/provider.py:660]: Actually, sorry, the previous comment may have been confusing, let me rephrase. I think we **do** need to call the `put_rest_api(..)` method directly, however, ideally with an updated `context` variable. 

Maybe we could pull out the util code `local_context = create_aws_request_context(..)` which creates a custom context inside `_call_moto(..)`, and make it reusable, such that we can pass a proper/updated context here?

Something like:
```
request = PutRestApiRequest(
    restApiId=response.get("id"),
    failOnWarnings=str_to_bool(fail_on_warnings) or False,
    parameters=parameters or {},
    body=io.BytesIO(body_data),
)
local_context = create_custom_context(context, "PutRestApi", request)
return self.put_rest_api(context, request)
```

Comment (calvernaz) [reply localstack/services/apigateway/provider.py:660]: So I went down that path and did the same for creation. The import operation is a combination of create+put. I extracted the function and created a new context. Along the way learned that account_id and the headers need to be set on the new context, something that before wasn't happening on `_call_moto` because it was based on the original context; it adds a bit of minimum overhead but nothing major.

For the record, I don't think the new context was necessary given how the put_rest_api operation is coded since it is not making a call to the backend (moto), so no "dispatching" is happening.

**Comment by whummer** [comment]
Looks like there is still a small test issue with the snapshot test - can you please check @calvernaz ?
```
>> match key: import_rest_api
	#x1B[31m(-)#x1B[0m /binaryMediaTypes/[0] ( '*/*' )
	#x1B[32m(+)#x1B[0m /binaryMediaTypes/[0] ( not present )
	#x1B[32m(+)#x1B[0m /tags ( {} )

	Ignore list (please keep in mind list indices might not work and should be replaced):
	["$..binaryMediaTypes", "$..tags"]
```

File: tests/integration/apigateway/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import os
+
+# parent directory of this file
+PARENT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+OPENAPI_SPEC_PULUMI_JSON = os.path.join(PARENT_DIR, "files", "openapi.spec.pulumi.json")
```
Comment (whummer) [inline tests/integration/apigateway/__init__.py:5]: nit: we usually keep the `__init__.py` files clean of custom logic. Not critical, as this is just a test module, but something to consider for future iterations. 👍

**Review by whummer — APPROVED** [review]
LGTM! 👍 (once the tests are green..) Great to see this fixed @calvernaz
